core/src/main/java/greencity/mapping/EcoNewsCommentDtoMapper.java

```diff
@@ -3,24 +3,35 @@
 import greencity.dto.econewscomment.EcoNewsCommentDto;
 import greencity.dto.econewscomment.EcoNewsCommentAuthorDto;
 import greencity.entity.EcoNewsComment;
+import greencity.entity.enums.CommentStatus;
 import org.modelmapper.AbstractConverter;
 import org.springframework.stereotype.Component;
 
 @Component
 public class EcoNewsCommentDtoMapper extends AbstractConverter<EcoNewsComment, EcoNewsCommentDto> {
     @Override
     protected EcoNewsCommentDto convert(EcoNewsComment ecoNewsComment) {
-        return EcoNewsCommentDto.builder()
-            .id(ecoNewsComment.getId())
-            .text(ecoNewsComment.getText())
-            .modifiedDate(ecoNewsComment.getModifiedDate())
-            .author(EcoNewsCommentAuthorDto.builder()
-                .id(ecoNewsComment.getUser().getId())
-                .name(ecoNewsComment.getUser().getName())
-                .userProfilePicturePath(ecoNewsComment.getUser().getProfilePicturePath())
-                .build())
-            .likes(ecoNewsComment.getUsersLiked().size())
-            .currentUserLiked(ecoNewsComment.isCurrentUserLiked())
-            .build();
+        EcoNewsCommentDto dto = new EcoNewsCommentDto();
+        dto.setId(ecoNewsComment.getId());
+        dto.setModifiedDate(ecoNewsComment.getModifiedDate());
+        if (ecoNewsComment.isDeleted()) {
+            dto.setStatus(CommentStatus.Deleted);
+            return dto;
+        }
+
+        if (ecoNewsComment.getCreatedDate().isEqual(ecoNewsComment.getModifiedDate())) {
```

Comment:
Do I understand correctly that `CommentStatus.Deleted` will be always overridden by `CommentStatus.Original` or `CommentStatus.Edited`?